src/main/java/com/liferay/ide/idea/server/LiferayDockerServerDebuggerRunner.java

```diff
@@ -0,0 +1,247 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.ide.idea.server;
+
+import com.intellij.build.BuildView;
+import com.intellij.debugger.DebugEnvironment;
+import com.intellij.debugger.DebuggerManager;
+import com.intellij.debugger.DebuggerManagerEx;
+import com.intellij.debugger.DefaultDebugEnvironment;
+import com.intellij.debugger.engine.DebugProcess;
+import com.intellij.debugger.engine.DebugProcessImpl;
+import com.intellij.debugger.impl.DebuggerSession;
+import com.intellij.debugger.impl.GenericDebuggerRunnerSettings;
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.RemoteConnection;
+import com.intellij.execution.configurations.RunProfile;
+import com.intellij.execution.configurations.RunProfileState;
+import com.intellij.execution.configurations.RunnerSettings;
+import com.intellij.execution.executors.DefaultDebugExecutor;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.execution.runners.ExecutionEnvironment;
+import com.intellij.execution.ui.ExecutionConsole;
+import com.intellij.execution.ui.RunContentDescriptor;
+import com.intellij.openapi.application.Application;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.externalSystem.service.execution.ExternalSystemRunnableState;
+import com.intellij.openapi.externalSystem.service.execution.ExternalSystemTaskDebugRunner;
+import com.intellij.openapi.project.Project;
+import com.intellij.util.net.NetUtils;
+
+import java.net.ServerSocket;
+
+import java.util.Collection;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * @author Simon Jiang
+ */
+public class LiferayDockerServerDebuggerRunner
+	extends ExternalSystemTaskDebugRunner implements ILiferayDockerServerRunnerCallback {
+
+	@Override
+	public boolean canRun(@NotNull String executorId, @NotNull RunProfile runProfile) {
+		if (DefaultDebugExecutor.EXECUTOR_ID.equals(executorId) &&
+			(runProfile instanceof LiferayDockerServerConfiguration)) {
+
+			return true;
+		}
+
+		return false;
+	}
+
+	@NotNull
+	@Override
+	public String getRunnerId() {
+		return "LiferayDockerServerDebugger";
+	}
+
+	@Nullable
+	@Override
+	protected RunContentDescriptor createContentDescriptor(
```

Comment:
Why do we have to override this method?  If we were able to avoid overriding this method, we could delete the ForkedDebuggerThread.java file